src/main/java/com/starrocks/connector/flink/table/source/StarRocksDynamicLookupFunction.java

```diff
@@ -0,0 +1,131 @@
+import com.starrocks.connector.flink.table.source.struct.SelectColumn;
+
+import org.apache.flink.table.data.GenericRowData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.functions.FunctionContext;
+import org.apache.flink.table.functions.TableFunction;
+import org.apache.flink.types.Row;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StarRocksDynamicLookupFunction extends TableFunction<RowData> {
+    
+    private static final Logger LOG = LoggerFactory.getLogger(StarRocksDynamicLUFunction.class);
+    
+    private final ColunmRichInfo[] filterRichInfos;
+    private final StarRocksSourceOptions sourceOptions;
+    private QueryInfo queryInfo;
+    private final SelectColumn[] selectColumns;
+    private final List<ColunmRichInfo> columnRichInfos;
+    private List<StarRocksSourceDataReader> dataReaderList;
+    
+    private final long cacheMaxSize;
+    private final long cacheExpireMs;
+    private final int maxRetryTimes;
+
+    // cache for lookup data
+    private Map<Row, List<RowData>> cacheMap;
+
+    private transient long nextLoadTime;
+
+    public StarRocksDynamicLookupFunction(StarRocksSourceOptions sourceOptions, 
+                                          ColunmRichInfo[] filterRichInfos, 
+                                          List<ColunmRichInfo> columnRichInfos,
+                                          SelectColumn[] selectColumns
+                                          ) {
+        this.sourceOptions = sourceOptions;
+        this.filterRichInfos = filterRichInfos;
+        this.columnRichInfos = columnRichInfos;
+        this.selectColumns = selectColumns;
+
+        this.cacheMaxSize = sourceOptions.getLookupCacheMaxRows();
+        this.cacheExpireMs = sourceOptions.getLookupCacheTTL();
+        this.maxRetryTimes = sourceOptions.getLookupMaxRetries();
+
+        this.dataReaderList = new ArrayList<>();
+        this.cacheMap = new HashMap<>();
+        this.nextLoadTime = -1L;
+    }
+    
+    @Override
+    public void open(FunctionContext context) throws Exception {
+        super.open(context);
+    }
+
+    public void eval(Object... keys) {
+        reloadData();
+        Row keyRow = Row.of(keys);
+        List<RowData> curList = cacheMap.get(keyRow);
+        if (curList != null) {
+            curList.parallelStream().forEach(data -> {
+                collect(data);
+            });
+        }
+    }
+
+    private void reloadData() {
+
+        if (nextLoadTime > System.currentTimeMillis()) {
+            return;
+        }
+        if (nextLoadTime > 0) {
+            LOG.info("Lookup join cache has expired after {} (ms), reloading", this.cacheExpireMs);
+        } else {
+            LOG.info("Populating lookup join cache");
+        }
+        cacheMap.clear();
+        List<RowData> tmpDataList = new ArrayList<>();
+        String SQL = "select * from " + sourceOptions.getDatabaseName() + "." + sourceOptions.getTableName();
```

Comment:
make the databaseName and tableName back quoted;
use stringbuilder instead of `+`